[FSDP][state_dict] Create a FSDP/EP unittest

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115009

This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

fegin committed:
[FSDP][state_dict] Create a FSDP/EP unittest

This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

[ghstack-poisoned]

fduwjj reviewed (approved):
LGTM and please make linter happy.

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc H-Huang awgu kwen2501 wanchaol fduwjj wz337 wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc H-Huang awgu kwen2501 wanchaol fduwjj wz337 wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc H-Huang awgu kwen2501 wanchaol fduwjj wz337 wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 reviewed (approved):
LGTM! 

Could probably re-run the distributed test and the device_mesh import should be working.

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 kiukchung d4l3k lucasllc

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 kiukchung d4l3k lucasllc

[ghstack-poisoned]

fegin committed:
Update on "[FSDP][state_dict] Create a FSDP/EP unittest"


This PR create a unittest to simulate FSDP + EP case.

Differential Revision: [D51507280](https://our.internmc.facebook.com/intern/diff/D51507280/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 kiukchung d4l3k lucasllc

[ghstack-poisoned]

fegin commented:
something is wrong with ghexport
